src/renderer/services/dex.js

```diff
@@ -284,21 +284,31 @@ export default {
           .then((res) => {
             const orders = res.data.orders;
             orders.forEach((order) => {
-              const marketForOrder = _.filter(store.state.markets, (market) => {
+              const marketForOrder = _.find(store.state.markets, (market) => {
                 return market.marketName === order.marketName;
               });
 
-              if (!marketForOrder || marketForOrder.length === 0) {
+              if (!marketForOrder) {
                 return;
               }
 
               if (order.side === 'Buy') {
-                order.assetIdToGive = marketForOrder[0].baseAssetId;
+                order.assetIdToGive = marketForOrder.baseAssetId;
                 order.quantityToGive = order.price * order.quantity;
               } else {
-                order.assetIdToGive = marketForOrder[0].quoteAssetId;
+                order.assetIdToGive = marketForOrder.quoteAssetId;
                 order.quantityToGive = order.quantity;
               }
+
+              const localOrder = _.find(store.state.orderHistory, (searchOrder) => {
+                return order.orderId === searchOrder.orderId;
+              });
+
+              if (localOrder && localOrder.status === 'Cancelling'
+                && moment.utc().diff(localOrder.cancelledAt, 'milliseconds') < timeouts.CANCEL_ORDER) {
```

Comment:
It seems like this logic will cause the order that is cancelling to stay in the cancelling state until the timeout of 5 minutes is reached. Where does it cause the stated to go to cancelled if the order cancels before the timeout is up?

Reply:
You're right, this had a problem, wouldn't update when it came back as fully cancelled from the API. Re-worked this, with a new PR here: https://github.com/aphtoken/aph-wallet-gui/pull/126/files